cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsPipelineSpecGenerator.java

```diff
@@ -96,9 +102,41 @@ public DataStreamsPipelineSpec generateSpec(DataStreamsConfig config) throws Val
     configureStages(config, specBuilder);
     //Configure the at least once processing mode for this pipeline
     configureAtleastOnceMode(config, specBuilder);
+    //Configure retries
+    configureRetries(specBuilder);
     return specBuilder.build();
   }
 
+  @VisibleForTesting
+  void configureRetries(DataStreamsPipelineSpec.Builder specBuilder) {
+    Long maxRetryTimeInMins = getFromRuntimeArgs(Constants.CDAP_STREAMING_MAX_RETRY_TIME_IN_MINS);
```

Comment:
nit: it's a little cleaner to pass in the default as `getFromRuntimeArgs(keyname, defaultValue)`. Then the method can return a long and you don't have to worry about nulls.

Reply:
Done.